File: leetcode/majority_element.cpp
using namespace std;
#include <vector>
#include <iostream>
#include <unordered_map>



class Solution {
public: 
    int majorityElement(vector<int>& nums) {
        // "&" is used to pass the vector "nums" by reference to the function
        int n = nums.size();
        unordered_map<int, int> m; 

        for (int i = 0; i < n; i++) {
            m[nums[i]]++;
        }

        n = n/2;
        for (auto x : m) {
            if (x.second > n) {
                return x.first;
            }
        }

        return 0;
    }
};

int main() {
    Solution s;
    vector<int> nums = {2, 3, 3, 3, 4, 5, 3};
    cout << s.majorityElement(nums) << endl;
}

File: leetcode/majority_element.py
from collections import defaultdict
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        m = defaultdict(int)
        
        for num in nums:
            m[num] += 1

        n = n // 2
        for key, value in m.items():
            if value > n:
                return key
            
        return 0
    
if __name__ == "__main__":
    s = Solution()
    nums = [2, 3, 3, 3, 4, 5, 3]
    print(s.majorityElement(nums))